### PR TITLE
Use datagouv/components without build

### PR DESCRIPTION
**Pull request by nicolaskempf57**
`@datagouv/components` now exports 3 files :
- `@datagouv/components` the js bundled
- `@datagouv/components/dist/style.css` the css bundled
- `@datagouv/components/ts` the source `index.ts`

udata-front is now using the source directly instead of relying on the build version.
This doesn't change anything for the end-user, everything is bundled during udata-front build.

Most of the changes are imports moved from `@datagouv/components` to `@datagouv/components/ts`.

I moved 2 JS files to TS because it was easier to do so than to try to see if vite handle TS import in JS file.

I know that `useUserAvatar` should be moved to a `helper` but I didn't want to clutter this PR.

### CODE REVIEW

File: udata_front/theme/gouvfr/assets/js/helpers.ts
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/js/helpers.ts:1]: Removed imports here were unused and the `_` is added to follow typescript convention with unused variable

File: udata_front/theme/gouvfr/assets/less/style.less
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/assets/less/style.less:1]: Compiled css isn't necessary anymore and both the components and udata-front CSS is generated with vite in a single file

File: udata_front/theme/gouvfr/datagouv-components/src/components/ReuseCard/ReuseCard.vue
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/datagouv-components/src/components/ReuseCard/ReuseCard.vue:1]: unused import

File: udata_front/theme/gouvfr/datagouv-components/src/composables/useComponentsForHook.ts
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/datagouv-components/src/composables/useComponentsForHook.ts:1]: This composable was used when the preview was outside of the resource and this is not the case anymore

File: udata_front/theme/gouvfr/datagouv-components/src/helpers/i18n.ts [hunk 1]
```diff
@@ -1,11 +1,10 @@
 import dayjs from "dayjs/esm";
 import LocalizedFormat from 'dayjs/esm/plugin/localizedFormat';
 import RelativeTime from "dayjs/esm/plugin/relativeTime";
-import { DefineLocaleMessage, I18n, Locale, createI18n } from "vue-i18n";
+import { type I18n, createI18n } from "vue-i18n";
 import messages from '@intlify/unplugin-vue-i18n/messages';
 import { config } from "../config";
 import { api_root_absolute } from "../api/api";
-import { getRegisteredTranslations } from "@etalab/udata-front-plugins-helper";
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/datagouv-components/src/helpers/i18n.ts:8]: everything related to udata-front-plugins-helper is unused so it's removed.

removing udata-front-plugins-helper is the next step but it significant enough to be a separate PR

File: vite.config.dev.ts [hunk 2]
```diff
@@ -43,8 +46,10 @@ async function getConfig(): Promise<UserConfig> {
         hook: 'writeBundle'
       }),
     ],
+    envDir: "udata_front/theme/gouvfr/datagouv-components",
     build: {
       minify: false,
+      cssCodeSplit: false,
```
Comment (nicolaskempf57) [inline vite.config.dev.ts:52]: We don't want css splitting because we are doing the import ourself without manifest

File: vite.config.dev.ts [hunk 3]
```diff
@@ -43,8 +46,10 @@ async function getConfig(): Promise<UserConfig> {
         hook: 'writeBundle'
       }),
     ],
+    envDir: "udata_front/theme/gouvfr/datagouv-components",
```
Comment (nicolaskempf57) [inline vite.config.dev.ts:49]: We are using the `.env` file from the components folder otherwise the components source are missing expected variables

File: udata_front/theme/gouvfr/datagouv-components/src/helpers/index.ts [hunk 4]
```diff
@@ -39,14 +39,14 @@ export const filesize = (val: number) => {
   return `${val.toFixed(1)}Y${suffix}`
 }
 
-export const summarize = (val: number) => {
+export const summarize = (val: number, fractionDigits = 0) => {
```
Comment (nicolaskempf57) [inline udata_front/theme/gouvfr/datagouv-components/src/helpers/index.ts:42]: This was the test I did to check that the modification is showing on udata-front

File: udata_front/theme/gouvfr/datagouv-components/package.json [hunk 5]
```diff
@@ -7,10 +7,22 @@
...
+      "import": "./dist/data-gouv-fr-components.js"
+    },
+    "./ts": "./src/index.ts",
+    "./dist/style.css": "./dist/style.css"
+  },
   "module": "./dist/data-gouv-fr-components.js",
   "types": "./dist/data-gouv-fr-components.d.ts",
   "scripts": {
     "dev": "vite",
+    "dev:udata_front": "vite build --watch",
```
Comment (maudetes) [inline udata_front/theme/gouvfr/datagouv-components/package.json:25]: What is this for?

**Review by maudetes — COMMENTED** [review]
Thank you a lot for the clean PR explanation! I am reviewing this PR early for once :stuck_out_tongue: 
Indeed, I think we can start with this one before adding more refactoring.

I have some questions as I stumble on this:
* Other datagouv-components users should update their import path as well?
* I am unsure of the datagouv components release process now?

I think we can update udata-front & datagouv-components Readme files.

File: udata_front/theme/gouvfr/datagouv-components/package.json (hunk 5 above)
Comment (nicolaskempf57) [reply udata_front/theme/gouvfr/datagouv-components/package.json:25]: Oops, this is a leftover of a previous trial

**Review by ThibaudDauce — APPROVED** [review]
:100:

**Review by maudetes — APPROVED** [review]
:ok_hand:
